cmp/grid/GridModel.js

```diff
@@ -716,14 +731,16 @@ export class GridModel {
     /**
      * Autosize columns to fit their contents.
      *
-     * @param {string|string[]} [colIds] - which columns to autosize; defaults to all leaf columns.
+     * @param {GridAutosizeOptions} options - overrides of default autosize options to use for this autosize.
      *
      * This method will ignore hidden columns, columns with a flex value, and columns with
      * autosizing disabled via the autosizeOptions.enabled flag.
      */
-    autosizeColumns(colIds) {
-        colIds = colIds ?? this.getLeafColumns().map(col => col.colId);
+    async autosizeColumnsAsync(options = {}) {
+        options = {...this.autosizeOptions, ...options};
 
+        const {columns} = options;
+        let colIds = isFunction(columns) ? columns() : columns ?? this.getLeafColumns().map(col => col.colId);
```

Comment:
wouldn't it be simpler/safer to have this function just be a predicate function that would be applied to a leaf column and return true if the column should be resized?  

We always want to operate on some subset of the leaf columns